src/test/java/io/temporal/samples/hello/HelloActivityTest.java

```diff
@@ -73,7 +74,10 @@ public void testActivityImpl() {
 
   @Test
   public void testMockedActivity() {
-    GreetingActivities activities = mock(GreetingActivities.class);
+    // withoutAnnotations() is required to stop Mockito from copying
+    // method-level annotations from the GreetingActivities interface
+    GreetingActivities activities =
+        mock(GreetingActivities.class, withSettings().withoutAnnotations());
```

Comment:
What's the reason that you don't want Mockito to preserve the annotations? 
Why if you do preserve them (testMockedActivityWithoutSettings) you expect the exception?

Just asking because if you have an Activity method with let's say

    @ActivityMethod(name = "sayHello")
    String composeGreeting(String greeting, String name);

The activity type is different if you remove the annotation, and you probably would want to test with the 
actually registered type from the annotation name param? Same really with the @ActivityInterface annotation.

Just wondering.

Reply:
The annotation on the interface class `GreetingActivities` should not be affected. The `withoutAnnotations()` setting removes annotations that Mockito copies from the interface class to the dynamically generated mock. Without it, the `worker.registerActivitiesImplementations(activities)` call fails, as it [explicitly checks](https://github.com/temporalio/sdk-java/blob/v1.0.7/temporal-sdk/src/main/java/io/temporal/common/metadata/POJOActivityImplMetadata.java#L65-L70) that the activity implementation class should not have `@ActivityMethod` annotations.

Someone has reported that behavior as a https://github.com/temporalio/sdk-java/issues/332 issue for the Java SDK.